packages/x-components/tests/unit/base-modal.spec.ts

```diff
@@ -0,0 +1,153 @@
+import { mount } from '@cypress/vue';
+import Vue from 'vue';
+import BaseModal from '../../src/components/modals/base-modal.vue';
+import { BaseXBus } from '../../src/plugins/x-bus';
+import { XPlugin } from '../../src/plugins/x-plugin';
+import { e2eAdapter } from '../../src/adapter/e2e-adapter';
+
+/**
+ * Renders an {@link BaseModal} component with the provided options.
+ *
+ * @param  options - Options to render {@link BaseModal} with.
+ * @returns Helper methods for the rendered {@link BaseModal}.
+ */
+function renderBaseModal({
+  clientHeaderSelector = '',
+  clientHeaderHeight = {},
+  template = `
+      <div class="wrapper">
+        <div class="header"></div>
+        <BaseModal :clientHeaderSelector="clientHeaderSelector" :open="open" />
+      </div>
+  `
+}: RenderBaseModalOptions = {}): RenderBaseModalAPI {
+  XPlugin.resetInstance();
+
+  mount(
+    {
+      components: {
+        BaseModal
+      },
+      template,
+      props: ['clientHeaderSelector', 'open']
+    },
+    {
+      vue: Vue.extend({}),
+      plugins: [[new XPlugin(new BaseXBus()), { adapter: e2eAdapter }]],
+      propsData: {
+        clientHeaderSelector,
+        open: true
+      },
+      style: `
+        body {
+          margin: 0;
+        }
+        .header {
+          height: ${clientHeaderHeight.desktop}px;
+        }
+        @media only screen and (min-width: 320px) and (max-width: 480px) {
+          .header {
+            height: ${clientHeaderHeight.mobile}px;
+          }
+        }
+        @media screen and (min-width: 481px) and (max-width: 1024px){
+          .header {
+            height: ${clientHeaderHeight.tablet}px;
+          }
+        }
+        `
+    }
+  );
+
+  return {
+    getBaseModal() {
+      return cy.getByDataTest('modal');
+    }
+  };
+}
+
+describe('Testing base modal', () => {
+  it('places the modal at top if there is not a selector', () => {
+    const { getBaseModal } = renderBaseModal();
+    getBaseModal().should($modal => expect($modal.position().top).to.be.eq(0));
+  });
+
+  it('places the modal under the selector passed by prop', () => {
+    const options = {
+      clientHeaderHeight: { desktop: 100 },
+      clientHeaderSelector: '.header'
+    };
+    const { getBaseModal } = renderBaseModal(options);
+    getBaseModal().should($modal =>
+      expect($modal.position().top).to.be.eq(options.clientHeaderHeight.desktop)
+    );
+  });
+
+  it('updates position when resizing', () => {
+    const options = {
+      clientHeaderHeight: {
+        mobile: 100,
+        tablet: 400,
+        desktop: 200
+      },
+      clientHeaderSelector: '.header'
+    };
+    const { getBaseModal } = renderBaseModal(options);
+    // Starts in desktop
```

Comment:
I'd prefer to change this code comments in favour of logs, so we are also aware of this changes when seeing the test in action.

```suggestion
    cy.log('Viewport in desktop resolution');
```